source/ncengine/graphics2/frontend/subsystem/LightSubsystem.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "LightRenderState.h"
+#include "graphics2/ShaderTypes.h"
+#include "ncengine/ecs/EcsFwd.h"
+
+#include "ncengine/graphics/DirectionalLight.h"
```

Comment:
You have `EcsFwd` here, so you should be able to also forward declare all component types.

Reply:
Ah - I had forward declared them but then had to fully declare them when I added the vectors, but forgot to remove `EcsFwd.h`